include/cuda/std/stream

```diff
@@ -0,0 +1,156 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA_STREAM
+#define _CUDA_STREAM
+
+#include "cstddef"
+
+#include "detail/__config"
+#include "detail/__pragma_push"
+#include "version"
+
+#include <cuda_runtime_api.h>
+#include <memory>
+#include <stdexcept>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_CUDA
+
+/**
+ * @brief A non-owning wrapper for a `cudaStream_t`.
+ *
+ */
+class stream_view {
+
+  /**
+   * @brief Constructs a `stream_view` of the "default" CUDA stream.
+   *
+   * For behavior of the default stream,
+   * see https://docs.nvidia.com/cuda/cuda-runtime-api/stream-sync-behavior.html
+   *
+   */
+  constexpr stream_view() = default;
+
+  /**
+   * @brief Constructs a `stream_view` from a `cudaStream_t` handle
+   *
+   * @note: It is the callers responsibilty to ensure the `stream_view` does not
+   * outlive the stream identified by the `cudaStream_t` handle.
+   *
+   */
+  explicit constexpr stream_view(cudaStream_t stream) : __stream{stream} {}
```

Comment:
This being explicit will make user code unnecessarily verbose - I think that disallowing construction from `int` and `nullptr_t` is enough to avoid ambiguity.

Reply:
@griwes agrees that this should be implicit.